Go/emitter/simple.go

```diff
@@ -0,0 +1,22 @@
+package emitter
+
+type Listener = func(...interface{})
+type EventEmitter map[string][]Listener
+
+func (ee EventEmitter) On(name string, listener Listener) {
+	var newListeners []Listener
```

Comment:
You can replace lines 7-13 with this line:
```go
	ee[name] = append(ee[name], listener)
```
You are doing a lot of redundant work here.